sw/device/lib/base/mmio.h

```diff
@@ -234,6 +234,62 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Masked Register field.
+ */
+typedef struct mmio_region_field32 {
+  uint32_t mask;  /**< The field mask. */
+  uint32_t index; /**< The field position within the a register. */
+  uint32_t value; /**< The field value to be written. */
+} mmio_region_field32_t;
+
+/**
+ * Sets the |field| from the MMIO region |base| at the given offset.
+ *
+ * This function performs a non-atomic read-write-modify operation on a
+ * MMIO region.
```

Comment:
I think we could do with a fuller explanation of this function than this. Something to include is a cross reference to `mmio_region_nonatomic_set_fields32` which will do multiple fields.

Reply:
good point.

Reply:
Done.